tests/test_eclsimple1dinterpolant.cpp

```diff
@@ -0,0 +1,1493 @@
+
+    // Left of range's left end-point.
+    {
+        const auto x = -1.0;
+
+        const auto pt =
+            PP::LocalInterpPoint::identify(abscissas, x);
+
+        BOOST_CHECK(pt.cat == Opm::Interp1D::PointCategory::LeftOfRange);
+
+        // Left of range's interval is always 0 (multiplied by .size() to
+        // enforce correct integer type in '==' check within Boost.Test).
+        BOOST_CHECK_EQUAL(pt.interval, 0*abscissas.size());
+
+        // t = x - xmin.
+        BOOST_CHECK_CLOSE(pt.t, x - abscissas.front(), 1.0e-10);
+    }
+
+    // Right of range's right end-point.
+    {
+        const auto x = 2.0;
+
+        const auto pt =
+            PP::LocalInterpPoint::identify(abscissas, x);
+
+        BOOST_CHECK(pt.cat == Opm::Interp1D::PointCategory::RightOfRange);
+
+        // Right of range's interval is always one less than the number of
+        // abscissas.
+        BOOST_CHECK_EQUAL(pt.interval, abscissas.size() - 1);
+
+        // t = x - xmax.
+        BOOST_CHECK_CLOSE(pt.t, x - abscissas.back(), 1.0e-10);
+    }
+
+    // Single abscissa point (left end point).
+    {
+        const auto x = abscissas[0];
+
+        const auto pt =
+            PP::LocalInterpPoint::identify(abscissas, x);
+
+        // We should identify this as being in range.
+        BOOST_CHECK(pt.cat == Opm::Interp1D::PointCategory::InRange);
+
+        // Interval should correspond to left (only) end-point.
+        BOOST_CHECK_EQUAL(pt.interval, 0*abscissas.size());
+
+        // t = x - abscissas[interval] (== 0.0).
+        BOOST_CHECK_CLOSE(pt.t, 0.0, 1.0e-10);
+    }
+
+    // Single abscissa point (right end point).
+    {
+        const auto x = abscissas[1];
+
+        const auto pt =
+            PP::LocalInterpPoint::identify(abscissas, x);
+
+        // We should identify this as being in range.
+        BOOST_CHECK(pt.cat == Opm::Interp1D::PointCategory::InRange);
+
+        // Interval should correspond to left end-point.
+        BOOST_CHECK_EQUAL(pt.interval, 0*abscissas.size());
+
+        // t = x - abscissas[interval] (== 1.0).
+        BOOST_CHECK_CLOSE(pt.t, 1.0, 1.0e-10);
+    }
+
+    // Common case: Points within range
+    {
+        const auto x = std::vector<double> {
+            0.1, 0.2, 0.4, 0.5, 0.75, 0.8,
+            1.0 - std::numeric_limits<double>::epsilon(),
+        };
+
+        for (const auto& xi : x) {
+            const auto pt =
+                PP::LocalInterpPoint::identify(abscissas, xi);
+
+            // We should identify this as being in range.
+            BOOST_CHECK(pt.cat == Opm::Interp1D::PointCategory::InRange);
+
+            // Interval should correspond to left end-point.
+            BOOST_CHECK_EQUAL(pt.interval, 0*abscissas.size());
+
+            // t = x - abscissas[interval] (== xi).
+            BOOST_CHECK_CLOSE(pt.t, xi, 1.0e-10);
+        }
+    }
+}
+
+BOOST_AUTO_TEST_CASE (MultipleIntervals)
+{
+    const auto abscissas = []() -> std::vector<double>
```

Comment:
This lambda did not compile for me:
```
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/iterator:959:14: error: cannot cast from lvalue of type 'const double' to rvalue reference
      type 'reference' (aka 'double &&'); types are not compatible
      return static_cast<reference>(*__i);
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/memory:1586:67: note: in instantiation of member function
      'std::__1::move_iterator<std::__1::__wrap_iter<const double *> >::operator*' requested here
                construct(__a, _VSTD::__to_raw_pointer(__begin2), *__begin1);
                                                                  ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/vector:1023:21: note: in instantiation of function template specialization
      'std::__1::allocator_traits<std::__1::allocator<double> >::__construct_range_forward<std::__1::move_iterator<std::__1::__wrap_iter<const double *>
      >, double *>' requested here
    __alloc_traits::__construct_range_forward(__a, __first, __last, this->__end_);
                    ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/vector:1171:9: note: in instantiation of function template specialization
      'std::__1::vector<double, std::__1::allocator<double> >::__construct_at_end<std::__1::move_iterator<std::__1::__wrap_iter<const double *> > >'
      requested here
        __construct_at_end(__first, __last, __n);
        ^
/Users/atgeirr/opm-build-debug/opm-flowdiagnostics-applications/tests/test_eclsimple1dinterpolant.cpp:341:16: note: in instantiation of function template
      specialization 'std::__1::vector<double, std::__1::allocator<double> >::vector<std::__1::move_iterator<std::__1::__wrap_iter<const double *> > >'
      requested here
        return {
               ^
1 error generated.
```

I cannot fix it, as I do not understand the code. I think the lambda is to const-initialize without having pvdg and nrwo leak into surrounding scope, but I do not see what you are trying to do here. If all you want is a vector of doubles, surely there is a simpler way?

Reply:
I'm tempted to call that a bug in libc++.  Could you try to remove the `std::make_move_iterator()` calls?

Reply:
I did but then that test fails:
```
5: unknown location:0: fatal error: in "PointLocationBinning/MultipleIntervals": std::invalid_argument: Cannot Relate Point to Non-Existent Range of Variable
5: /Users/atgeirr/opm-build-debug/opm-flowdiagnostics-applications/tests/test_eclsimple1dinterpolant.cpp:334: last checkpoint: "MultipleIntervals" test entry
```

Reply:
Scratch that, I erred when removing the make_move_iterator.